# Hedging Against Turkish Inflation

Selin Acar, Necati Furkan Çolak, Denis Lasser

# Introduction

The ongoing economic crisis in Türkiye has been characterized by high
inflation, the depreciation of the Turkish Lira (TRY), rising borrowing
costs, and increasing loan defaults. The underlying causes of the crisis
can be attributed to several factors, including political instability,
global economic pressures, and the implementation of unconventional
economic policies. Since 2018, inflation has been a persistent issue,
partly due to fluctuations in global oil prices and exchange rates, with
Turkey's inflation rate ranking among the highest in emerging markets. The effectiveness of inflation-targeting policies
has been limited by both internal shocks, such as political events and
the COVID-19 pandemic, and external factors, such as rising global
inflation. Furthermore, the Turkish
government's policy of maintaining low interest rates to stimulate
growth, despite the conventional theory that raising interest rates
should reduce inflation, has contributed significantly to the current
economic challenges. Nevertheless, interest
rates have been raised significantly. As of November 2024, the most
recent rate hike was in March 2024, when the central bank increased the
policy rate by 5 percentage points to 50\% as the year-on-year inflation
reached 75\%. Since then, the rate has remained
unchanged as the central bank has been committed to fighting high
inflation and stabilising the Turkish lira. As of October 2024,
year-on-year inflation stands at 48\%. This study aims
to analyze which assets, as well as which portfolio of assets, would
have been the most effective in hedging against Turkish inflation.

# Setup

In [1]:
import pandas as pd

# Reproducing the Research

If you want to reproduce the research you have to follow these steps:

1. Retrieve the raw data as described in the section "Data" below and place it in the directory `data/raw`.

2. Run the notebook [Database_Preprocess_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Database_Preprocess_Notebook.ipynb) to process the data

3. Run the notebook [Markowitz_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Markowitz_Notebook.ipynb) to run the Markowitz optimization

4. Run the notebook [Visuals_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Visuals_Notebook.ipynb) to create the visuals used in the guide, paper, and presentation.

If you are not interested in the detailed steps, you can just run the notebooks (option "run all cells") and read through the guide to get a general understandig and be presented with the results.


# Methodology

This study examines the effectiveness of various assets as hedges against Turkish inflation, using monthly data from January 2018 to September 2024. The monthly frequency was selected for the following
reasons:

1. The use of monthly data allows for the smoothing of short-term fluctuations. This facilitates the identification of more coherent relationships between inflation and asset returns.
2. Key economic indicators, including inflation rates, are typically reported on a monthly basis. By aligning asset data with this frequency, the analysis maintains consistency and relevance in relation to reported inflation rates.
3. The influence of short-term trading anomalies, seasonal effects and temporary market shocks on weekly data must be acknowledged. Monthly data minimizes the impact of these irregularities, thereby providing a more stable basis for the analysis of assets as potential inflation hedges.

## Data

This study includes a range of assets to capture different potential
hedging mechanisms against inflation.

Following indices were sourced from Bloomberg:

1. **TUCXEFYY Index**: This index tracks the Turkish CPI (Consumer
  Price Index) excluding categories such as food, energy, alcohol, and
  tobacco. This represents the core inflation rate and excludes the most
  volatile components of the CPI, given that the aforementioned
  categories are often subject to short-term price fluctuations due to
  external factors. In the context of hedging against inflation, it is
  more meaningful to focus on core inflation, as short-term fluctuations
  in volatile categories can distort the true underlying inflation rate.
  Therefore, this index serves as the benchmark and allows for direct
  comparison of the other assets against inflation.
2. **XU100 Index** (BIST 100): This index comprises the 100 largest
  companies traded at the Borsa Istanbul and provides insight into the
  response of Turkish stocks to inflation.
3. **GTTRY10YR** (10-Year Turkish Government Bond): Long-term
  government bonds, such as the 10-year Turkish government bond, are
  traditionally considered low-risk investments. Including this bond in
  the analysis allows for an assessment of the inflation risk premium
  embedded in Turkish government debt.
4. **XAU**: This index represents the price of one ounce of gold
  quoted in US Dollars (USD). Gold is often seen as a traditional hedge
  against inflation due to its historical retention of value. Analyzing
  gold prices in USD allows for an assessment of gold's ability to
  preserve purchasing power during periods of high inflation in Türkiye,
  while avoiding the potential distortions caused by exchange rate
  volatility.
4. **XBT BGN**: Bitcoin is included in the analysis as an
  alternative inflation hedge, with prices analyzed in USD. Analyzing
  Bitcoin in USD focuses on its performance as a global inflation hedge,
  independent of potential distortions caused by exchange rate
  volatility. This approach provides insights into Bitcoin's ability to
  preserve purchasing power and its relevance in the context of Turkish
  inflation.
5. **USGG10YR Index**: The 10-year US Government bond yield was used as the risk-free rate in the Markowitz Optimization. This rate serves as a benchmark for risk-free returns and providess a baseline to evaluate the performance of other assets and calculate excess returns in the portfolio construction.


Furthermore, the house price index data was obtained from the [Turkish
Central Bank's EVDS database](https://evds2.tcmb.gov.tr/index.php?/evds/portlet/KSAktmjeowo\%3D/en)
Specifically, the **TP KFE TR-1** series was used, which tracks
nationwide residential property price change in Türkiye. Real estate is
often considered a strong inflation hedge, as property values and rental
income tend to rise with inflation. This index allows an analysis of the
effectiveness of the real estate sector in preserving purchasing power
during inflationary periods.

### Sample Data

**Note about using your own data:** To use your own input data, please match the format of the raw data samples provided in the `data/sample data` directory. Once you've ensured that, you can simply copy your data into the data/raw directory and proceed with the guide without issues.

The detailed data processing can be seen in this notebook: [Database_Preprocess_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Database_Preprocess_Notebook.ipynb)


Samples for the data format can be found in the `data/sample data` directory. The raw data should have the following format (please note that the samples are fictional):

In [2]:
# for stock exchange data
df_stocks = pd.read_excel("data/sample data/borsa_ist.xlsx")
df_stocks

Date  Last Price (TRY)
0 2024-09-30           1234.56
1 2024-08-30           1234.56
2 2024-07-31           1234.56
3 2024-06-28           1234.56
4 2024-05-31           1234.56

In [3]:
# for bitcoin
df_btc = pd.read_excel("data/sample data/btc.xlsx")
df_btc

Date  Last Price (USD)
0 2024-09-30           1234.56
1 2024-08-31           1234.56
2 2024-07-31           1234.56
3 2024-06-30           1234.56
4 2024-05-31           1234.56

In [4]:
# for gold
df_gold = pd.read_excel("data/sample data/gold_prices.xlsx")
df_gold

Date  Last Price (USD)
0 2024-09-30           1234.56
1 2024-08-30           1234.56
2 2024-07-31           1234.56
3 2024-06-28           1234.56
4 2024-05-31           1234.56

In [5]:
# for government bonds
df_bonds = pd.read_excel("data/sample data/gov_bonds.xlsx")
df_bonds

Date  Yield (%)
0 2024-09-30      12.34
1 2024-08-30      12.34
2 2024-07-31      12.34
3 2024-06-28      12.34
4 2024-05-31      12.34

In [6]:
# for house price index
df_hpi = pd.read_excel("data/sample data/hpi.xlsx")
df_hpi

Tarih  TP KFE TR-1
0  2024-09         1.23
1  2024-08         1.23
2  2024-07         1.23
3  2024-06         1.23
4  2024-05         1.23

In [3]:
# for 10y US government bond yield
df_rf = pd.read_excel("data/sample data/USGG10YR Index.xlsx")
df_rf

Date  Yield in %
0 2024-09-30        1.23
1 2024-08-30        1.23
2 2024-07-31        1.23
3 2024-06-28        1.23
4 2024-05-31        1.23

In [7]:
# for inflation
df_inf = pd.read_excel("data/sample data/Turkish Inflation 2018-2024.xlsx")
df_inf

Date   in %
0 2024-09-30  12.34
1 2024-08-31  12.34
2 2024-07-31  12.34
3 2024-06-30  12.34
4 2024-05-31  12.34

### Data Processing

Once the raw data is obtained and placed in the 'data/raw' directory, we can import and process the data. The data is first imported and cleaned. Then the real returns of the assets are calculated according to the formula described below in the subchapter **Real Returns**. Finally, all the data is gathered into one table and safed in a SQLite database for easy access.

For more detailed information on the data processing, see the following notebook: [Database_Preprocess_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Database_Preprocess_Notebook.ipynb)


## Analysis Techniques

The analysis consists of two main steps. First, the effectiveness of
each individual asset in providing a hedge is evaluated. Subsequently,
an optimal hedging portfolio is determined.

### Real Returns

The first step is to calculate the **real return** for each asset relative to the Turkish inflation rate. This ensures a consistent measure of an asset's effectiveness in preserving purchasing power in Türkiye's inflationary environment.

$$
    \text{Real Return} = \frac{1 + \text{Nominal Return}} {1 + \text{Inflation Rate}} - 1
$$

This formula adjusts the nominal return of each asset to account for the impact of Turkish inflation. By expressing returns in real terms, it provides a clearer picture of an asset's ability to act as an inflation hedge, irrespective of whether it is denominated in TRY or USD.

This first step evaluates each asset's ability to preserve purchasing
power. Assets with consistently positive excess returns are considered
to have stronger hedging characteristics, as they maintain or increase
purchasing power above inflation. This initial analysis provides a basis
for assessing each asset's stand-alone ability to serve as an inflation
hedge.

Here is a quick overview of the real returns calculated:

In [8]:
df_master = pd.read_csv("data/processed/master_table.csv")

columns = ["Date", 'real_return_bitcoin', 'real_return_gold', 'real_return_stockex', 'real_return_bond', 'real_return_hpi']

df_master[columns].iloc[1:11]

Date  real_return_bitcoin  real_return_gold  real_return_stockex  \
1   2024-07             0.033023          0.145895            -0.817967   
2   2024-06            -0.798180         -0.418346            -0.460427   
3   2024-05             1.467751         -0.292343            -0.366619   
4   2024-04            -0.924560         -0.232486            -0.068653   
5   2024-03             2.159272          0.619049             0.751348   
6   2024-02            47.672475         -0.405119            -0.652090   
7   2024-01            -0.421561         -0.488646             0.053929   
8   2023-12             1.434699         -0.315337             0.988190   
9   2023-11             0.645958         -0.194566            -0.790802   
10  2023-10            10.385174          0.374515            -0.085540   

    real_return_bond  real_return_hpi  
1          -0.165578        -0.262818  
2          -0.208981        -0.249867  
3          -0.262995        -0.074697  
4          -0.278203        -0.362104  
5          -0.288351        -0.339971  
6          -0.285429        -0.156453  
7          -0.271676        -0.205646  
8          -0.274930        -0.259088  
9          -0.263742        -0.359244  
10         -0.256136        -0.200725

### Markowitz Portfolio Optimization

Building on the individual asset analysis, the next step applies
**Markowitz portfolio optimization** to construct an optimal
portfolio that maximizes the hedging effectiveness against Turkish
inflation. Markowitz optimization, also referred to as
**Mean-Variance Optimization**, determines the portfolio
composition that achieves the highest return for a given level of risk /
the lowest risk for a given return. In the context of this study, the
aim is to construct a portfolio that minimizes inflation risk while
providing stable excess returns. By balancing a range of assets,this
optimization creates a portfolio that preserves purchasing power against
inflation more effectively than any individual asset alone.

The process involves the following calculations:

1. **Portfolio Return**
  $$
  \text{Portfolio Return} = R_p = \sum_{i=1}^{n} w_i R_i
  $$

where:

- $R_p$: Portfolio return
- $w_i$: Weight of asset $i$ in the portfolio
- $R_i$: Return of asset $i$
- $n$: Number of assets in the portfolio


2. **Portfolio Variance**
  $$\text{Portfolio Variance} = \sigma_p^2 = \sum_{i=1}^{n} \sum_{j=1}^{n} w_i w_j \sigma_{ij}$$

where:

- $\sigma_p^2$: Portfolio variance (measure of risk)
- $w_i, w_j$: Weights of assets $i$ and $j$ in the portfolio
- $\sigma_{ij}$: Covariance between the returns of assets $i$ and $j$
- $\sigma_{ii}$ or $\sigma_i^2$: Variance of asset $i$ (case when $i = j$)


3. **Minimizing Portfolio Risk**
  $$
  \min_{w} \, \sigma_p^2 \quad \text{subject to:} \quad \sum_{i=1}^{n} w_i R_i = R_{\text{target}}, \quad \sum_{i=1}^{n} w_i = 1, \quad w_i \geq 0
  $$

where:

- $\min_{w} \, \sigma_p^2$: The objective is to minimize the portfolio
  variance
- $\sum_{i=1}^{n} w_i R_i = R_{\text{target}}$: The portfolio must
  achieve a specified target return ($R_{\text{target}}$)
- $\sum_{i=1}^{n} w_i = 1$: The sum of portfolio weights must equal 1,
  this ensures that the portfolio is fully invested
- $w_i \geq 0$: Weights must be non-negative, assuming no short-selling

Solving the optimization problem generates the efficient frontier, which
represents the set of optimal portfolios that offer the highest return
for a given level of risk. The efficient frontier is derived by varying
$R_{\text{target}}$ and solving the optimization problem for each
level of return.

For more details on the code for the Markowitz optimization, please refer to this notebook: [Markowitz_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Markowitz_Notebook.ipynb)

# Results

For analyizing the results of the analysis, we'll refer to visuals created by the [Visuals_Notebook.ipynb](https://github.com/denis-lasser/hedging-against-turkish-inflation/blob/main/notebooks/Visuals_Notebook.ipynb) notebook.

## Real Return of Individual Assets

### Turkish Stocks

![](reports/figures/real_stock_returns.png)

Turkish stocks are highly volatile, with several sharp peaks and periods of low or negative returns. For instance, there are noticeable spikes in returns, especially around mid/late 2020 and mid-2022. Therefore, Turkish equities appear to offer opportunities for inflation hedging during certain periods, but are inconsistent as a hedge. While they may reflect market optimism during inflationary periods, they still carry significant risks, as shown by the high volatility.

### Turkish Government Bond

![](reports/figures/real_bond_returns.png)

10 year Turkish government bonds show prolonged periods of negative real returns, especially as inflation rises. Recent improvements in real returns may reflect monetary tightening, but the overall trend suggests that they struggle to keep up with inflation. Therefore, Turkish bonds are not a good hedge against inflation due to their sensitivity to high inflation and credit risk.

### Gold

![](reports/figures/real_gold_returns.png)

Gold has traditionally been seen as an inflation hedge. Despite some negative returns, gold maintains a conistent performance compared to other assets. This makes gold is a modest hedge against Turkish inflation, as it tends to hold its value during inflationary spikes. 

### Bitcoin

![](reports/figures/real_bitcoin_returns.png)

Bitcoin's real returns are highly volatile. It occasionally generates high positive real returns but also shows frequent periods of near-zero or negative returns. Therefore, it is not a consistent hedge against Turkish inflation due to its high volatility. While it offers occasional large positive returns, it also carries significant downside risk. This makes it speculative rather than a stable store of value in the context of inflation hedging.

### Turkish Real Estate

![](reports/figures/real_hpi_returns.png)

Real returns on Turkish real estate show occasional spikes, but generally remain close to zero, with also a few periods of slightly negative returns. The peak in real returns occurred in late 2021, likely driven by a spike in inflation, with real estate acting as a safe haven. Consequently, real estate appears to be a viable hedge against inflation during periods of high inflation in Türkiye. The sharp increase in real returns suggests strong investor demand for real assets to preserve wealth amid economic uncertainty.


### Comparison

Comparing the assets, one can conclude for the period from January 2018 to September 2024: 

- **Least Effective Hedges:**
    - **Turkish Bonds:** Struggle to generate positive real returns under high inflation.
    - **Bitcoin:** High volatility reduces reliability as a consistent inflation hedge.

    
- **Balanced Hedge:**
    - **Stocks:** Provide a mix of potential growth and inflation protection but require careful risk management due to volatility.

    
- **Best Inflation Hedges:**
    - **Gold:** Offers steady value preservation, especially during high inflation periods.
    - **House Price Index:** Reasonable alignment with inflation but lacks significant outperformance.


## Markowitz Portfolio Optimization

Portfolio allocation using the mean real return of the US 10y Gov Bond (-20.87\%):

- **Turkish Stocks**: 3.48\%
- **Turkish Government Bond**: 57.76\%
- **Gold**: 8.40\%
- **Bitcoin**: 0.15\%
- **Turkish Real Estate**: 30.21\%

Resulting in:

- **Expected Portfolio Return**: 1.34\%
- **Expected Portfolio Volatility**: 19.77\%

Portfolio allocation using the median real return of the US 10y Gov Bond (-13.71\%):

- **Turkish Stocks**: 7.71\%
- **Turkish Government Bond**: 0.00\%
- **Gold**: 24.67\%
- **Bitcoin**: 0.47\%
- **Turkish Real Estate**: 67.15\%

Resulting in:

- **Expected Portfolio Return**: 22.05\%
- **Expected Portfolio Volatility**: 41.25\%

The results highlight the sensitivity of portfolio allocation and expected outcomes to the chosen risk-free rate. It also emphasizes the importance of robust assumptions in the Markowitz optimization framework.

Using the average real return of the 10-year US Treasury (-20.87\%), the portfolio is heavily allocated to Turkish government bonds (57.76\%) and Turkish real estate (30.21\%), with minimal allocations to gold (8.40\%) and bitcoin (0.15\%). This allocation reflects a strategy to minimize risk while achieving a modest expected portfolio return (1.34\%) and relatively low volatility (19.77\%).

In contrast, using the median real return (-13.71\%) shifts the allocation dramatically, with a complete exclusion of Turkish government bonds (0.00\%) and a heavier weighting in gold (24.67\%) and Turkish real estate (67.15\%). This allocation results in a much higher expected return (22.05\%), but at the cost of much higher volatility (41.25\%).

The differences in portfolio allocations and outcomes illustrate that the choice between the mean and median real risk-free rate can profoundly affect optimization results, particularly in environments with volatile inflation and significant outliers. The mean risk-free rate, which is more sensitive to extreme values, drives a conservative allocation with lower returns and risk. The median risk-free rate, which is robust to outliers, leads to a more aggressive allocation that prioritizes higher expected returns at the expense of greater volatility. These results show the importance of conducting robustness checks, especially when inflation or other economic conditions vary significantly, as assumptions about the risk-free rate directly impact the portfolio's risk-return profile.

It is also worth noting that the Markowitz optimization is purely mathematical and does not take into account behavioral factors or practical constraints such as aversion to extreme volatility or liquidity needs. 


# Conclusion and Future Research

This study highlights the sensitivity of portfolio allocation to the choice of the real risk-free rate in inflation-adjusted Markowitz optimization. Turkish real estate emerges as a key portfolio component due to its relatively high real returns and alignment with inflationary trends. However, the resulting portfolio achieves high expected returns at the cost of significant volatility, particularly when the median real risk-free rate is applied.

Future research could look into areas such as: 
- Expand the asset universe to include more complex financial products, such as derivatives (e.g., inflation-linked swaps or options), exchange-traded funds (ETFs), or foreign-denominated assets. These products could provide more effective and flexible hedging mechanisms against inflation.
- Explore dynamic portfolio strategies that adapt to changing inflationary environments.
- Assess the role of alternative risk measures, such as downside risk or drawdown, to create portfolios that better address practical investor concerns.


# Generating Files for Publication


If you wish to create the paper or presentation in a presentation ready format, here are some instructions to convert the .tex files to .pdf.

The recommended way is to use the `texlive/texlive` distribution with docker. For this purpose, the medium distribution is enough. To pull the image using the docker CLI (command line interface):

    docker pull texlive/texlive:lastest-medium

Alternatively, you can also use docker's graphical interface. To run the docker container, run this command from the `reports` directory (if you run it from a different directory, you will have to navigate to the correct path inside the container):

    docker run -v $(pwd):/workdir -it texlive/texlive:latest-medium bash

From inside the container run the following commands after each other. Note, that it's best to run the last command twice, as is shown.

**For the paper:**

    pdflatex Hedging-Against-Turkish-Inflation-Paper.tex
    bibtex Hedging-Against-Turkish-Inflation-Paper.aux
    pdflatex Hedging-Against-Turkish-Inflation-Paper.tex
    pdflatex Hedging-Against-Turkish-Inflation-Paper.tex

**For the presentation:**

    pdflatex Hedging-Against-Turkish-Inflation-Presentation.tex
    bibtex Hedging-Against-Turkish-Inflation-Presentation.aux
    pdflatex Hedging-Against-Turkish-Inflation-Presentation.tex
    pdflatex Hedging-Against-Turkish-Inflation-Presentation.tex

You can now find the .pdf of both the paper and the presentation in the `reports` directory.
